admin-client/src/main/java/com/epam/admin_client/service/LoginService.java

```diff
@@ -0,0 +1,42 @@
+package com.epam.admin_client.service;
+
+import com.epam.test_generator.controllers.user.request.LoginUserDTO;
+import com.epam.test_generator.dto.TokenDTO;
+import javax.annotation.Resource;
+import org.springframework.core.env.Environment;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class LoginService {
+
+
+    private final String URI_LOGIN;
```

Comment:
LOGIN_API_URL

I see problem that if certain environment variables were not set - application will start up normally, but user will keep getting errors on UI (but no errors in logs at all).

To prevent this, application should fail to start when required parameters were not set.
First approach is use environment.getRequiredProperty("propname")
Second approach is to inject properties into constructor of bean and call Validate.notNull(property)

Reply:
btw ALL_CAPITALS names are usually only given to static final variables, except loggers.